.github/workflows/smoke-test.yml

```diff
@@ -0,0 +1,127 @@
+            export DIFF=$( git diff --name-only ${{ github.event.before }} $GITHUB_SHA )
+            echo "Diff between ${{ github.event.before }} and $GITHUB_SHA"
+          fi
+          echo "$DIFF"
+          # Escape newlines (replace \n with %0A)
+          echo "::set-output name=diff::$( echo "$DIFF" | sed ':a;N;$!ba;s/\n/%0A/g' )"
+
+      - name: Set matrix for smoke-tests
+        id: set-matrix
+        run: |
+          DIFF="${{ steps.diff.outputs.diff }}"
+          JSON="{\"include\":["
+
+          # Loop by lines
+          while read path; do
+            # Set $directory to substring before /
+            directory="$( echo $path | cut -d'/' -f1 -s )"
+
+            if [ "$directory" == .github ]; then
+              continue # Exclude .github directory
+            elif [ "$directory" == deploy ]; then
+              continue # Exclude deploy directory
+            elif [ "$directory" == samples ]; then
+              continue # Exclude samples directory
+            fi
+
+            # Add build to the matrix only if it is not already included
+            JSONline="{\"directory\": \"$directory\"},"
+            if [[ "$JSON" != *"$JSONline"* ]]; then
+              JSON="$JSON$JSONline"
+            fi
+          done <<< "$DIFF"
+
+          # Remove last "," and add closing brackets
+          if [[ $JSON == *, ]]; then
+            JSON="${JSON%?}"
+          fi
+          JSON="$JSON]}"
+          echo $JSON
+
+          # Set output
+          echo "::set-output name=matrix::$( echo "$JSON" )"
+
+
+  smoke-tests:
+    runs-on: ubuntu-latest
+    name: Run Smoke Tests on "${{ matrix.directory }}" chart
+    needs: generate-matrix
+    strategy:
+      matrix: ${{fromJson(needs.generate-matrix.outputs.matrix)}}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Git Checkout
+      
+      - name: Create Kind Cluster
+        uses: helm/kind-action@v1.1.0
+        with:
+          config: .github/kind.yml
+      
+      - name: Install helm
+        uses: Azure/setup-helm@v1.1
+        with:
+          version: v3.5.3
+      
+      - name: Deploy "${{ matrix.directory }}" chart
+        run: |
+          kubectl create namespace ${{ env.NAMESPACE }}
+          kubectl apply -f deploy/dependencies/mongo.yaml -n ${{ env.NAMESPACE }}
+          kubectl apply -f deploy/dependencies/redis.yaml -n ${{ env.NAMESPACE }}
+
+          if [ "${{ matrix.directory }}" == tyk-headless ]; then
+            helm install ${{ matrix.directory }} ./${{ matrix.directory }} -n ${{ env.NAMESPACE }} --timeout ${{ env.TIMEOUT }}
+          elif [ "${{ matrix.directory }}" == tyk-hybrid ]; then
+            # Need to set proper orgId, apiKey and connString values
+            helm install --set gateway.rpc.rpcKey=5cc03283d07e7f00019404b3 --set gateway.rpc.apiKey=ChAnGeMe --set gateway.rpc.connString=localhost ${{ matrix.directory }} ./${{ matrix.directory }} -n ${{ env.NAMESPACE }} --timeout ${{ env.TIMEOUT }}
```

Comment:
`rpcKey` and `apiKey` should be secrets. It will need a permanent Tyk Cloud deployment it could connect too though — please raise an internal ticket (`TD`) to set one up.

Reply:
Unblock yourself - split between what you can and cannot do without a permanent deployment, release the former and hang the later on the  ticket.

Reply:
The entire hybrid chart can't be tested without permanent deployment. I mean can set dummy values like i did, but that's not the point of having smoke-tests.

I believe it's better to knock it off in a single ticket rather than coming back to it in sometime with a different ticket.